test/integration/lib/expression_viz_service_test.rb

```diff
@@ -0,0 +1,44 @@
+require "test_helper"
+
+class ExpressionVizServiceTest < ActiveSupport::TestCase
+
+  setup do
+    @user = FactoryBot.create(:user)
+    @basic_study = FactoryBot.create(:detached_study, name: 'Basic Viz')
+    @basic_study_cluster_file = FactoryBot.create(:study_file,
+                                                  name: 'cluster1.txt',
+                                                  file_type: 'Cluster',
+                                                  study: @basic_study)
+    @basic_study_exp_file = FactoryBot.create(:study_file,
+                                                  name: 'dense.txt',
+                                                  file_type: 'Expression Matrix',
+                                                  study: @basic_study)
+    @cluster_group = FactoryBot.create(:cluster_group_with_cells,
+                                       study_file: @basic_study_cluster_file,
+                                       cell_data: {
+                                         x: [1, 4 ,6],
+                                         y: [7, 5, 3],
+                                         cells: ['A', 'B', 'C']
+                                       },
+                                       annotations: [{name: 'foo', type: 'group', values: ['bar', 'bar', 'baz']}])
```

Comment:
`ClusterGroup` documents have a field called `cell_annotations`, not `annotations`.  Also, the `values` array holds _unique_ values, so it cannot have repeated entries.

Reply:
this is correct.  The `cluster_group_with_cells` factory defines a transient property 'annotations', which it then processes to create both the annotation data array, and the cell_annotations property you mention

Reply:
Ah - understood.  Thanks for the clarification.